### PR TITLE
Fix logic around URL parsing, add hierarchy drill down

### PR DESCRIPTION
**Pull request by browniefed**
## Change description

First part of hierarchy. Add simple drill down logic for both Teams and  Domain.
Fixes URL param parsing logic

https://www.loom.com/share/af515f871999475b886e864bcaf04c35

## Type of change

- [ ] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [x] Improvement (improves codebase without changing functionality)

### Development

- [x] The changelog has been updated as appropriate
- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

**Review by FoodProduct — COMMENTED** [review]
1. Missing cursor/hover state on drill-downable things
2. URL update logic doesn't care if drill-down is actually feasible

https://github.com/cortexapps/backstage-plugin/assets/3069614/1edeb144-3da9-4c97-99d5-aa18da58c6da

File: src/components/ReportsPage/HeatmapPage/HeatmapPage.tsx [hunk 1]
```diff
@@ -69,13 +72,14 @@ export const HeatmapPage = () => {
     useHierarchy: !!searchParams.has('hierarchy'),
     hideWithoutChildren: !searchParams.has('hideWithoutChildren'),
     scoreFilters: {
-      serviceIds: searchParams.get('serviceIds')?.split(',').map((i) => parseInt(i, 10)).filter(isFinite) ?? defaultFilters.scoreFilters.serviceIds,
-      groups: searchParams.get('groups')?.split(',') ?? defaultFilters.scoreFilters.groups,
-      teams: searchParams.get('teams')?.split(',') ?? defaultFilters.scoreFilters.teams,
-      users: searchParams.get('users')?.split(',') ?? defaultFilters.scoreFilters.users,
-      domainIds: searchParams.get('domainIds')?.split(',').map((i) => parseInt(i, 10)).filter(isFinite) ?? defaultFilters.scoreFilters.domainIds,
-      levels: searchParams.get('levels')?.split(',') ?? defaultFilters.scoreFilters.levels,
+      serviceIds: searchParams.getAll('serviceIds').map((i) => parseInt(i, 10)).filter(isFinite) ?? defaultFilters.scoreFilters.serviceIds,
```
Comment (FoodProduct) [inline src/components/ReportsPage/HeatmapPage/HeatmapPage.tsx:75]: 👍

File: src/components/ReportsPage/HeatmapPage/Tables/SingleScorecardHeatmapTable.tsx [hunk 2]
```diff
@@ -39,8 +52,34 @@ interface SingleScorecardHeatmapTableProps {
   useHierarchy: boolean;
   hideWithoutChildren: boolean;
   domainTagByEntityId: Record<string, string[]>;
+  setFiltersAndNavigate: Dispatch<React.SetStateAction<HeatmapPageFilters>>;
+  filters: HeatmapPageFilters;
 }
 
+type HierarchNode = DomainHierarchyNode | TeamHierarchyNode;
```
Comment (FoodProduct) [inline src/components/ReportsPage/HeatmapPage/Tables/SingleScorecardHeatmapTable.tsx:59]: 👀 
```suggestion
type HierarchyNode = DomainHierarchyNode | TeamHierarchyNode;
```

**Comment by browniefed** [comment]
@FoodProduct Ah yep, I'll fix those in this PR, but I forgot to mention this isn't finalized by any means.
I'm getting as much logic as I can done so then also Pavel can do some work over night. So doing this little by litte.

**Review by FoodProduct — COMMENTED** [review]
Besides the functional things I noticed, code changes generally LGTM. Other random notes

1. Thanks for the formatting -- we really should enforce this (there's a ticket for it somewhere...)
2. Our current lack of code coverage in this repo is 😿 but the onus in not on you to fix in this PR 🤷

**Comment by browniefed** [comment]
Updated checks for if we're in hierarchy mode, and also added link buttons so things are clickable

**Review by FoodProduct — APPROVED** [review]
I'm honestly not 100% sure how to create an actual hierarchy for testing but this generally LGTM

**Pull request merged**